src/Swarm/Game/Step.hs

```diff
@@ -122,24 +119,116 @@ gameTick = do
           Just (v, s) -> do
             replStatus .= REPLWorking (Typed (Just v) ty req)
             baseRobot . robotContext . defStore .= s
-          Nothing -> return ()
-        _otherREPLStatus -> return ()
-    Nothing -> return ()
+          Nothing -> pure ()
+        _otherREPLStatus -> pure ()
+    Nothing -> pure ()
 
-  -- Possibly update the view center.
-  modify recalcViewCenter
+  when ticked $ do
```

Comment:
Why do we only do this when `ticked` is true?  In practice, it seems like this means when single-stepping a robot that is executing `move`, I first see the robot move when the `move` command is executed, and then I have to hit Ctrl-O to step a few more times before the view center updates.  Is there a particular reason for doing it this way?

It's actually particularly confusing since the view center updates at the same time that all the other robots get a chance to move etc., which makes it hard to see what is happening.  I would rather have the view center update at the same time that the focused robot moves, then you can more easily see what all the other robots are doing.

Also, don't we want to check the win condition immediately when the currently focused robot does something, rather than waiting for the end of the tick?  Otherwise I imagine it would be confusing when single stepping your robot and it e.g. does something that achieves the objective but nothing happens until you do a few more steps.

Reply:
The view centre could be recalculated at every step, I will fix that.

Win condition **should not** be checked on every step because:
1. It is slow.
2. That is not how it works for normal time mode, defeating the purpose of debugging.

Reply:
Yes, I see, that makes sense.